libs/oeth/swap/src/actions/swapCurveSfrxeth.ts

```diff
@@ -0,0 +1,268 @@
+import { contracts, tokens } from '@origin/shared/contracts';
+import { prepareWriteContractWithTxTracker } from '@origin/shared/providers';
+import {
+  ETH_ADDRESS_CURVE,
+  isNilOrEmpty,
+  subtractSlippage,
+} from '@origin/shared/utils';
+import {
+  erc20ABI,
+  getAccount,
+  getPublicClient,
+  prepareWriteContract,
+  readContract,
+  writeContract,
+} from '@wagmi/core';
+import { formatUnits } from 'viem';
+
+import { GAS_BUFFER } from '../constants';
+
+import type {
+  Allowance,
+  Approve,
+  EstimateAmount,
+  EstimateApprovalGas,
+  EstimateGas,
+  EstimateRoute,
+  Swap,
+} from '@origin/shared/providers';
+
+const config = {
+  routes: [
+    tokens.mainnet.OETH.address,
+    contracts.mainnet.OETHCurvePool.address,
+    ETH_ADDRESS_CURVE,
+    '0xa1f8a6807c402e4a15ef4eba36528a3fed24e577',
+    tokens.mainnet.frxETH.address,
+    tokens.mainnet.sfrxETH.address,
+    tokens.mainnet.sfrxETH.address,
```

Comment:
no idea how this works, but do we mean to have this one twice?

Reply:
yes, this is how curve params are prepared, it means "call stake function on sfrxETH to wrap frxETH". Great times figuring that out 😓